examples/snap-plugin-collector-rand/rand/rand.go

```diff
@@ -54,8 +55,23 @@ var (
 	}
 )
 
+var req bool = false
+
 func init() {
 	rand.Seed(42)
+
+	//The required-config flag is added for testing plugin-lib-go.
+	//When the flag is set, an additional policy will be added in GetConfigPolicy().
+	//This additional policy has a required field. This simulates
+	//the situation when a plugin requires a config to load.
+	plugin.AddFlag(
+		cli.BoolFlag{
+			Name:        "required-config",
```

Comment:
hello @jcooklin, I know that this flag was introduced to simulate a situation that config policy defines required configuration. I am just thinking what benefits it brings in comparison with adding such required `value` for metric "string/static" without this flag and all this stuff related to handling and manage that. 

My suggestion is to do that similar to how it's done in regular collector plugin - I would like to present the most real situation, not a fake one - I mean there is no plugin with flag `required-config`. What do you think about that?